File: 028.implement-strstr/implement-strstr.cs
public class Solution {
    public int StrStr(string haystack, string needle) {
        int l1 = haystack.Length;
        int l2 = needle.Length;
        if (l2 == 0) return 0;
        if (l2 > l1) return -1;
        for (int i = 0; i <= l1 - l2; i++)
        {
            if(needle.CompareTo(haystack.Substring(i, l2)) == 0) return i;
        }
        return -1;
    }
}

File: 0028.implement-strstr/implement-strstr.java
class Solution {
    public int strStr(String haystack, String needle) {
        int l1 = haystack.length();
        int l2 = needle.length();
        if (l2 == 0) return 0;
        if (l2 > l1) return -1;
        return haystack.indexOf(needle);
    }
}